### PR TITLE
Javascript tasks 3,4 and 5

### PR DESCRIPTION
**Pull request by AnchalAgrawal4**
Added tasks 3,4 and 5.
Please review and let me know for any changes.

### CODE REVIEW

File: JavaScript/Task3/ClosuresAndScopes.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+//Solution 2:
+function createFunctions(n) {
+  var callbacks = [];
+
+  for (var i = 0; i < n; i++) {
+    (function() {
+      var j = i;
+      callbacks.push(function() {
+        console.log(j);
```
Comment (SaiTeja2904) [inline JavaScript/Task3/ClosuresAndScopes.js]: Remove Console logs

File: JavaScript/Task5/extract-nested-object-reference.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+//http://www.codewars.com/kata/extract-nested-object-reference
+
+Object.prototype.hash = function(string) {
+  var properties = string.split(".");
+  var myObj = this;
+  for (let i = 0; i < properties.length; i++) {
+    !!myObj && (myObj = myObj[properties[i]]);
```
Comment (SaiTeja2904) [inline JavaScript/Task5/extract-nested-object-reference.js]: Explore Boolean(myObj)

File: JavaScript/Task5/extract-nested-object-reference.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+//http://www.codewars.com/kata/extract-nested-object-reference
+
+Object.prototype.hash = function(string) {
+  var properties = string.split(".");
+  var myObj = this;
```
Comment (SaiTeja2904) [inline JavaScript/Task5/extract-nested-object-reference.js:5]: Don't use var, instead use let or const depending upon the situation

File: JavaScript/Task3/CanYouKeepASecret.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+//http://www.codewars.com/kata/can-you-keep-a-secret
+
+function createSecretHolder(secret) {
+  return {
+    getSecret: function() {
```
Comment (SaiTeja2904) [inline JavaScript/Task3/CanYouKeepASecret.js:5]: Use anonymous functions

File: JavaScript/Task3/ClosuresAndScopes.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+//http://www.codewars.com/kata/closures-and-scopes/train/javascript
+
+//Solution 1:
+function createFunctions(n) {
+  let callbacks = [];
+
+  for (let i = 0; i < n; i++) {
+    callbacks.push(function() {
+      return i;
```
Comment (SaiTeja2904) [inline JavaScript/Task3/ClosuresAndScopes.js]: Can be replaced with "callbacks.push(_ => i);"

File: JavaScript/Task3/ClosuresAndScopes.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
+//http://www.codewars.com/kata/closures-and-scopes/train/javascript
+
+//Solution 1:
+function createFunctions(n) {
+  let callbacks = [];
```
Comment (SaiTeja2904) [inline JavaScript/Task3/ClosuresAndScopes.js]: Can use const as you are not reassigning the value but changing the array!

File: JavaScript/Task5/ArrayHelpers.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+Array.prototype.square = function() {
+  return this.map(number => number ** 2);
+};
+
+Array.prototype.cube = function() {
+  return this.map(number => number ** 3);
+};
+
+Array.prototype.average = function() {
+  return this.sum() / this.length;
```
Comment (SaiTeja2904) [inline JavaScript/Task5/ArrayHelpers.js:13]: This function throws an error if I call this method over an empty array ie length 0. So add edge cases also.

**Review by SaiTeja2904 — CHANGES_REQUESTED** [review]

File: JavaScript/Task3/CanYouKeepASecret.js (hunk 4 above)
Comment (AnchalAgrawal4) [reply JavaScript/Task3/CanYouKeepASecret.js:5]: Can you please tell me how to do it? I'm a not able to understand how to use anonymous function in case.

**Review by SaiTeja2904 — APPROVED** [review]

File: JavaScript/Task3/CanYouKeepASecret.js (hunk 4 above)
Comment (SaiTeja2904) [reply JavaScript/Task3/CanYouKeepASecret.js:5]: getSecret : () => {}

Comment (AnchalAgrawal4) [reply JavaScript/Task3/CanYouKeepASecret.js:5]: done

**Pull request merged**